prow/cmd/deck/rerun.go

```diff
@@ -0,0 +1,371 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	prowapi "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	v1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	prowv1 "k8s.io/test-infra/prow/client/clientset/versioned/typed/prowjobs/v1"
+	"k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/gerrit/client"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/githuboauth"
+	"k8s.io/test-infra/prow/pjutil"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/plugins/trigger"
+)
+
+var (
+	// Stores the annotations and labels that are generated
+	// and specified within components.
+	ComponentSpecifiedAnnotationsAndLabels = sets.NewString(
+		// Labels
+		client.GerritRevision,
+		client.GerritPatchset,
+		client.GerritReportLabel,
+		github.EventGUID,
+		"created-by-tide",
+		// Annotations
+		client.GerritID,
+		client.GerritInstance,
+	)
+)
+
+// ProwJobEvent contains the minimum information required to start a ProwJob.
+type ProwJobEvent struct {
+	Name string `json:"name"`
```

Comment:
Agreed, in addition, removing the struct can make this function more usable in other parts of the code base (if needed to be exported) since a developer will not need to work with the arbitrary struct defined.